### ISSUE TITLE
Incorrect indent when using curly braces and switch case

### ISSUE THREAD

**Issue by Imarok** [eslint/eslint#8879]
**Tell us about your environment**

* **ESLint Version:** v3.19.0
* **Node Version:** v4.2.6
* **npm Version:** 3.5.2

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 6
    },
    "rules": {
        "no-cond-assign": 1,
        "no-constant-condition": 1,
        "no-dupe-args": 1,
        "no-dupe-keys": 1,
        "no-duplicate-case": 1,
        "no-empty": 1,
        "no-extra-boolean-cast": 0,
        "no-extra-parens": 0,
        "no-extra-semi": 1,
        "no-func-assign": 1,
        "no-negated-in-lhs": 1,
        "no-obj-calls": 1,
        "no-unreachable": 1,
        "use-isnan": 1,
        "valid-jsdoc": 0,
        "valid-typeof": 1,

        "block-scoped-var": 0,
        "consistent-return": 1,
        "default-case": 1,
        "dot-notation": 1,
        "no-else-return": 1,
        "no-invalid-this": 1,
        "no-lone-blocks": 1,
        "no-loop-func": 0,
        "no-multi-spaces": 1,
        "no-new": 1,
        "no-redeclare": 0,
        "no-return-assign": 1,
        "no-self-assign": 1,
        "no-self-compare": 1,
        "no-unmodified-loop-condition": 1,
        "no-unused-expressions": 1,
        "no-unused-labels": 1,
        "no-useless-concat": 0,

        "no-delete-var": 1,
        "no-label-var": 1,
        "no-shadow-restricted-names": 1,
        "no-shadow": 1,
        "no-undef": 0,
        "no-undef-init": 1,
        "no-unused-vars": 0,

        "new-cap": 0,
        "new-parens": 1,
        "no-unneeded-ternary": 1,

        "no-class-assign": 1,
        "no-const-assign": 1,
        "no-dupe-class-members" : 1,
        "prefer-const": 0,

        "space-in-parens": 1,
        "space-before-function-paren": ["warn", "never"],
        "space-infix-ops": 1,
        "space-unary-ops": 1,
        "spaced-comment": ["warn", "always"],
        "comma-spacing": 1,
        "semi-spacing": 1,
        "brace-style": ["warn", "allman", { "allowSingleLine": true }],
        "no-multi-assign": 1,
        "no-trailing-spaces": 1,
        "object-curly-spacing": ["warn", "always"],
        "operator-assignment": 1,
        "operator-linebreak": ["warn", "after"],
        "semi": 1,
        "yoda": 1,
        "curly": ["warn", "multi"],
        "quote-props": 1,

        "indent": ["warn", "tab"],
        "no-mixed-spaces-and-tabs": ["warn", "smart-tabs"]
    }
}


```

</details>

**The indent rule doesn't work properly, when using curly braces inside a switch case:**
This should be correct code with this config:
```js

function bar(msg)
{
	switch (n)
	{
	case 10:
		return 1;

	case 20:
		foo();

	case 40:
	{
		let i = n * 2;
		return i;
	}

	case 50:
	{
		let z = n * 2;
		n = foobar(n, z);
		break;
	}
	}
}
```
But it seems like the indentation have to be like this to be accepted by ESLint:
```js

function bar(msg)
{
	switch (n)
	{
	case 10:
		return 1;

	case 20:
		foo();

	case 40:
		{
			let i = n * 2;
			return i;
		}

	case 50:
		{
			let z = n * 2;
			n = foobar(n, z);
			break;
		}
	}
}
```

**Comment by not-an-aardvark** [comment]
Thanks for reporting. This is a known problem and was fixed in ESLint 4.x. You can run `npm install eslint@latest` to upgrade (it's possible you'll also have to modify some things -- see the [migration guide](http://eslint.org/docs/user-guide/migrating-to-4.0.0)).

**Comment by Imarok** [comment]
nice to know :)
Thank you very much for your fast reply.
When will ES become stable?
(Or why do I have still ESLint 3.19? I installed it just recently)

**Comment by not-an-aardvark** [comment]
It is already stable, but some popular shareable configs don't support it yet. So if you're using one of those or you used `eslint --init` and selected one of those, it's possible 3.x was installed.

**Comment by Imarok** [comment]
hmm, I think I just did `npm install eslint` and got 3.19
